### ISSUE TITLE
Hide the user's timezone until they've signed in for the first time (aka validated their account)

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#11458]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Problem:
Before user's validate their accounts and sign in for the first time, we default their timezone NVP to PDT since we have no idea where their client lives. 

This is an issue when a user opens a chat report with an unvalidated user, since there's a good chance that the `It's 9:00am for XXXX` text we display of the report action compose box is inaccurate.

## Solution:
Hide the `It's 9:00am for XXXX` text for unvalidated accounts.

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.9-0
**Reproducible in staging?:** 
**Reproducible in production?:** 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** https://user-images.githubusercontent.com/43996225/193163920-5e17fbf7-04ce-463d-b04c-958305a62918.mov
https://user-images.githubusercontent.com/43996225/193163930-8bdb0401-f071-4bc8-a46a-6abbc91c4643.mov
**Expensify/Expensify Issue URL:** 
**Issue reported by:** @chauchausoup
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1663949425736709

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lschurr** [comment]
Seems like this was agreed upon in Slack.

**Comment by francoisl** [comment]
Hi @chauchausoup, I'm not really understanding the proposed solution:

> Keep a dropdown in modal once a new chat user profile is clicked.

Could you expand a bit on it please? Also, in the Slack thread, it sounds like it was suggested [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1664380551427909?thread_ts=1663949425.736709&cid=C01GTK53T8Q) that we simply not show a user's timezone if it's not set?

**Comment by chauchausoup** [comment]
@francoisl actually two separate FRs (Feature Request) were merged here. 
[FR1](https://expensify.slack.com/archives/C01GTK53T8Q/p1663949425736709)
[FR2](https://expensify.slack.com/archives/C01GTK53T8Q/p1663950270131509)

The details if of FR2 but discussion is of FR1. 
Let me know if this helps your confusion.

**Comment by JmillsExpensify** [comment]
I don't agree with the solution, so I'm removing `External` and `Help Wanted` labels.

**Comment by JmillsExpensify** [comment]
A couple of reasons why I don't like this solution:

> For mobile phone profiles we can detect the time zone from country code 

This is not a reliable way to detect time zone. The US has six time zones for one country code, as an example

> Keep a dropdown in modal once a new chat user profile is clicked. This will solve issue with faulty time zone being shown in each chat box. 

I'm not sure what this means.

**Comment by JmillsExpensify** [comment]
If we do anything, then I think the most simple solution would be to simply not show a user's timezone unless a timezone has been set (cc @trjExpensify as we chatted about this in our 1:1).

**Comment by trjExpensify** [comment]
Yeah, additionally from the thread [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1664379967141129?thread_ts=1663949425.736709&cid=C01GTK53T8Q) this doesn't solve the problem for unvalidated email accounts that are yet to sign-in either: 

>Great observation! I think the solution doesn’t fully solve the problem though, because it will still exist for unvalidated email accounts you invite. The account’s timeZone NVP requires a value (IIRC from when we introduced this feature a few years ago on OldDot), so that’s why PST is the default set until they sign-in for the first time.

So I think the problem needs to be updated to something more reflective: 

**Problem:** The timeZone NVP is set to PDT by default until a user signs-in for the first time (at which point it updates automatically). That means that the timeZone being displayed in the chat is often incorrect until then, confusing and mislead the other chat participants, while introducing an element of doubt over the locale of that member.

>If we do anything, then I think the most simple solution would be to simply not show a user's timezone unless a timezone has been set

Yeah, a value is set in the timeZone NVP upon account creation, but not updated until they log-in. At which point it updates automatically (which is the default setting for setting your timezone). With that, should the solution be to only show a user's timeZone in the chat after they've validated their account/signed-in for the first time?

**Comment by JmillsExpensify** [comment]
> With that, should the solution be to only show a user's timeZone in the chat after they've validated their account/signed-in for the first time?

That's pretty simple. I like that.

**Comment by jasperhuangg** [comment]
Hey @JmillsExpensify @trjExpensify thank you guys for the discussion and clarification of what the actual issue is here. I've updated the OP to reflect your discussion. 

I can go ahead and knock this one out today so we don't have to wait for proposals. I think it might make things easier if the person implementing this has access to our internal code too.

**Comment by JmillsExpensify** [comment]
Cool, that sounds good. I'll keep myself assigned for the Upwork reporting / feature request payment.

**Comment by jasperhuangg** [comment]
PRs are all up and ready for review

**Comment by trjExpensify** [comment]
Speedy, nice!

**Comment by JmillsExpensify** [comment]
Linked PRs are still up and going through the review process.

**Comment by JmillsExpensify** [comment]
@chauchausoup Before we get this one merged, can you please apply for the Upwork job for reporting/getting this discussion started? Job is here: https://www.upwork.com/jobs/~011c0f1b36bcc1bd3c.

**Comment by chauchausoup** [comment]
> @chauchausoup Before we get this one merged, can you please apply for the Upwork job for reporting/getting this discussion started? Job is here: https://www.upwork.com/jobs/~011c0f1b36bcc1bd3c.

@JmillsExpensify done.

**Comment by JmillsExpensify** [comment]
I've hired @chauchausoup in the linked Upwork job for reporting. As for this issue, we're still waiting on one PR to merge, though it's on hold for a related issue.

**Comment by JmillsExpensify** [comment]
Removed the hold on the last remaining PR.

**Comment by JmillsExpensify** [comment]
Contributor has been paid out via Upwork. Keeping the issue closed.
